apps/web/server/routers/viewer/eventTypes.tsx

```diff
@@ -117,6 +117,7 @@ export const eventTypesRouter = createProtectedRouter()
 
       const data: Prisma.EventTypeCreateInput = {
         ...rest,
+        userId: teamId ? undefined : userId,
```

Comment:
AKAIK this is isn't needed when we're using the `connect` property from prisma. Also, we should have zod validations that prevents sending both `teamId` and `userId`

Reply:
> AKAIK this is isn't needed when we're using the connect property from prisma

As per my testing, `userId` isn't getting populated automatically, thus allowing the possibility to create multiple event-types with same slug for same user by bypassing the unique index.

Worth noting that `users` is of the type `User[]`. Prisma auto populates with connect property when there is a one-to-one or many-to-one mapping i.e. `user: { connect: ... }`

> Also, we should have zod validations that prevents sending both teamId and userId

`userId` is derived from user session so it's always available. It isn't passed as an input.  While `teamId` is passed in input optionally.